src/native/libs/Common/pal_atomic.h

```diff
@@ -4,7 +4,7 @@
 
 #pragma once
 
-#if defined(TARGET_UNIX)
+#if defined(TARGET_UNIX) || defined(TARGET_WASI)
```

Comment:
Browser too, for consistency ?

Reply:
I think the browser builds currently define `TARGET_UNIX`. I'd expect WASI to do the same but I opted for minimal changes.

Reply:
Browser has very good posix emulation, so you are right about `TARGET_UNIX`. I would still prefer more explicit reference.
WASI is not unix.

Reply:
While I understand your reasoning I hesitate to do the change to reduce collateral damage.

Prior to this PR the ICU shim was built in the Mono runtime and it set `TARGET_UNIX` for both for browser ([here](https://github.com/dotnet/runtime/blob/cabaab5e90e19256ae404970091350bb6f0f7139/src/mono/CMakeLists.txt#L696)) and WASI ([here](https://github.com/dotnet/runtime/blob/cabaab5e90e19256ae404970091350bb6f0f7139/src/mono/CMakeLists.txt#L691)). 

In the new setup the System.Globalization.Native library shares the same build system as System.Native. The existing System.Native build defines TARGET_UNIX for Browser, but not for WASI.

If I add `|| defined(TARGET_BROWSER)` here it will be strictly redundant (since `TARGET_BROWSER` implies `TARGET_UNIX`) and inconsistent with the rest of the System.Native code that has no such checks.

Reply:
I'm OK with this difference here for now, though we should probably move away from setting TARGET_UNIX for browser too. For example in the managed src/libraries build we do _not_ set `TargetsUnix` msbuild property which gets used to set `TARGET_UNIX` C# define for neither Browser nor Wasi.